corehq/ex-submodules/couchforms/util.py

```diff
@@ -119,92 +95,8 @@ def __exit__(self, exc_type, exc_val, exc_tb):
-    and format them uniformly, in place.
-
-    """
-    # this strips the timezone like we've always done
-    # todo: in the future this will convert to UTC
-    if isinstance(data, basestring) and re_loose_datetime.match(data):
-        try:
-            matching_datetime = iso8601.parse_date(data)
-        except iso8601.ParseError:
-            pass
-        else:
-            if phone_timezones_should_be_processed():
-                parent[key] = unicode(json_format_datetime(
-                    matching_datetime.astimezone(pytz.utc).replace(tzinfo=None)
-                ))
-            else:
-                parent[key] = unicode(json_format_datetime(
-                    matching_datetime.replace(tzinfo=None)))
-
-    elif isinstance(data, dict):
-        for key, value in data.items():
-            adjust_datetimes(value, parent=data, key=key)
-    elif isinstance(data, list):
-        for i, value in enumerate(data):
-            adjust_datetimes(value, parent=data, key=i)
-
-    # return data, just for convenience in testing
-    # this is the original input, modified, not a new data structure
-    return data
-
-
-def create_xform(xml_string, attachments=None, process=None):
-    """
-    create but do not save an XFormInstance from an xform payload (xml_string)
-    optionally set the doc _id to a predefined value (_id)
-    return doc _id of the created doc
-
-    `process` is transformation to apply to the form right before saving
-    This is to avoid having to save multiple times
-
-    If xml_string is bad xml
-      - raise couchforms.XMLSyntaxError
-
-    """
-    assert attachments is not None
-    json_form = convert_xform_to_json(xml_string)
-    adjust_datetimes(json_form)
-
-    _id = _extract_meta_instance_id(json_form) or XFormInstance.get_db().server.next_uuid()
-    assert _id
-
-    xform = XFormInstance(
-        # form has to be wrapped
-        {'form': json_form},
-        # other properties can be set post-wrap
-        _id=_id,
-        xmlns=json_form.get('@xmlns'),
-        received_on=datetime.datetime.utcnow(),
-    )
-    attachment_manager = AttachmentsManager(xform)
-
-    # Always save the Form XML as an attachment
-    attachment_manager.store_attachment('form.xml', xml_string, 'text/xml')
-
-    for name, filestream in attachments.items():
-        attachment_manager.store_attachment(name, filestream, filestream.content_type)
-
-    attachment_manager.commit()
-
-    # this had better not fail, don't think it ever has
-    # if it does, nothing's saved and we get a 500
-    if process:
-        process(xform)
-
-    lock = acquire_lock_for_xform(_id)
-    with ReleaseOnError(lock):
-        if _id in XFormInstance.get_db():
-            raise DuplicateError(xform)
-
-    return LockManager(xform, lock)
-
-
 def process_xform(instance, attachments=None, process=None, domain=None):
+    from corehq.form_processor.backends.couch.processor import FormProcessorCouch
```

Comment:
note: for now, this is grabbing it straight from the adapter instead of the interface. until we have enough infrastructure for it to really only return a `GenericXFormInstance`